### PR TITLE
Add workflows and workflow templates for PR decoration with test reports

### PR DESCRIPTION
**Pull request by turing85**
Proof of functionality:
- A PR without a `report-config.yml`:
  - PR: https://github.com/turing85/quarkus-artemis/pull/14
  - Initial comment is not posted: https://github.com/turing85/quarkus-artemis/actions/runs/10802614248
  - Build runs normally: https://github.com/turing85/quarkus-artemis/actions/runs/10802614245
  - Reports are not created: https://github.com/turing85/quarkus-artemis/actions/runs/10802711846
- A PR with `report-config.yml`:
  - PR: https://github.com/turing85/quarkus-artemis/pull/13
  - Initial comment is posted: https://github.com/turing85/quarkus-artemis/actions/runs/10802375811
  - Build runs normally: https://github.com/turing85/quarkus-artemis/actions/runs/10802375881
  - Reports are created: https://github.com/turing85/quarkus-artemis/actions/runs/10802531966

### CODE REVIEW

File: .github/workflows/perform-post-comment.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          then
+            present=true
+          fi
+          echo "config-present=${present}" >> ${GITHUB_OUTPUT}
+        env:
+          CONFIG_FILE: ${{ inputs.report-config-file }}
+      - name: Check report config is not empty
+        id: check
+        if: ${{ steps.lookup.outputs.config-present == 'true' }}
+        uses: mikefarah/yq@v4
```
Comment (gastaldi) [inline .github/workflows/perform-post-comment.yml]: `yq` is already present in the `ubuntu-latest` distribution

File: .github/workflows/perform-post-report.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          then
+            present="true"
+          fi
+          echo "config-present=${present}" >> ${GITHUB_OUTPUT}
+        env:
+          CONFIG_FILE: ${{ inputs.report-config-file }}
+      - name: Check report config is not empty
+        id: check
+        if: ${{ steps.lookup.outputs.config-present == 'true' }}
+        uses: mikefarah/yq@v4
```
Comment (gastaldi) [inline .github/workflows/perform-post-report.yml]: Same here

File: .github/workflows/perform-post-report.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        with:
+          cmd: yq '.reports | length > 0' < ${CONFIG_FILE}
+        env:
+          CONFIG_FILE: ${{ inputs.report-config-file }}
+      - name: Load report config file
+        id: load
+        if: ${{ steps.lookup.outputs.config-present == 'true' && steps.check.outputs.result == 'true' }}
+        uses: mikefarah/yq@v4
```
Comment (gastaldi) [inline .github/workflows/perform-post-report.yml]: And here

File: .github/report-config.sample.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+reports:
```
Comment (gastaldi) [inline .github/report-config.sample.yml]: Is this file essential? Why not moving its contents to report-config.yml?

Comment (turing85) [reply .github/report-config.sample.yml]: The idea was to provide a sample. The configuration is take from [`quarkus-artemis`](https://github.com/quarkiverse/quarkus-artemis/) which has an atypical. Do you have a suggestion for a more meaningful default? I was thinking about:
```yml
reports:
  - name: Surefire
    path: '**/target/surefie-reports/TEST*.xml'
    artifact-pattern: test-report*
  - name: Failsafe
    path: '**/target/failsafe-reports/TEST*.xml'
    artifact-pattern: test-report*
```

Comment (gastaldi) [reply .github/report-config.sample.yml]: That looks better, just noticed there is a typo in the surefire path ( `s/surefie/surefire`)

Comment (gastaldi) [reply .github/report-config.sample.yml]: How about Unit and Integration tests instead?

File: .github/workflows/perform-post-comment.yml (hunk 1 above)
Comment (turing85) [reply .github/workflows/perform-post-comment.yml]: Restructured.

File: .github/workflows/perform-post-report.yml (hunk 2 above)
Comment (turing85) [reply .github/workflows/perform-post-report.yml]: Restructured.

File: .github/workflows/perform-post-report.yml (hunk 3 above)
Comment (turing85) [reply .github/workflows/perform-post-report.yml]: Restructured.

File: .github/report-config.sample.yml (hunk 4 above)
Comment (turing85) [reply .github/report-config.sample.yml]: Done.

**Comment by turing85** [comment]
@gastaldi just for completeness: I updated the samples with fresh runs against the most recent version.

**Review by gastaldi — APPROVED** [review]

**Pull request merged**